### ISSUE TITLE
Why compile nginx from source instead of using official (signed) packages?

### ISSUE THREAD

**Issue by tarrenj** [bunkerity/bunkerized-nginx#3]
https://github.com/bunkerity/bunkerized-nginx/blob/14ec9f3e63383adc7ff30b79ee0318bf1f53db2f/compile.sh#L30

Nginx maintains packages for Alpine: http://nginx.org/en/linux_packages.html#Alpine

**Comment by DistractionRectangle** [comment]
Looks like it's necessary to bake in the [ModSecurity](https://github.com/SpiderLabs/ModSecurity-nginx#compilation), [headers more](https://github.com/openresty/headers-more-nginx-module#installation), [geoip](https://github.com/leev/ngx_http_geoip2_module#installing), and [cookie flag](https://github.com/AirisX/nginx_cookie_flag_module) modules

I was about a file a similar issue, but more so due to how the image is compiled. Running `nginx -V 2>&1 | tr ' ' '\n'` on 1.18.0/alpine (for arm32v7 in my case) yields 
```
nginx
version:
nginx/1.18.0
built
by
gcc
9.2.0
(Alpine
9.2.0)

built
with
OpenSSL
1.1.1g

21
Apr
2020
TLS
SNI
support
enabled
configure
arguments:
--prefix=/etc/nginx
--sbin-path=/usr/sbin/nginx
--modules-path=/usr/lib/nginx/modules
--conf-path=/etc/nginx/nginx.conf
--error-log-path=/var/log/nginx/error.log
--http-log-path=/var/log/nginx/access.log
--pid-path=/var/run/nginx.pid
--lock-path=/var/run/nginx.lock
--http-client-body-temp-path=/var/cache/nginx/client_temp
--http-proxy-temp-path=/var/cache/nginx/proxy_temp
--http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp
--http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp
--http-scgi-temp-path=/var/cache/nginx/scgi_temp
--with-perl_modules_path=/usr/lib/perl5/vendor_perl
--user=nginx
--group=nginx
--with-compat
--with-file-aio
--with-threads
--with-http_addition_module
--with-http_auth_request_module
--with-http_dav_module
--with-http_flv_module
--with-http_gunzip_module
--with-http_gzip_static_module
--with-http_mp4_module
--with-http_random_index_module
--with-http_realip_module
--with-http_secure_link_module
--with-http_slice_module
--with-http_ssl_module
--with-http_stub_status_module
--with-http_sub_module
--with-http_v2_module
--with-mail
--with-mail_ssl_module
--with-stream
--with-stream_realip_module
--with-stream_ssl_module
--with-stream_ssl_preread_module
--with-cc-opt='-Os
-fomit-frame-pointer'
--with-ld-opt=-Wl,--as-needed
```

vs running it on bunkerized-nginx
```
nginx
version:
nginx/1.18.0
built
by
gcc
9.3.0
(Alpine
9.3.0)

built
with
OpenSSL
1.1.1g

21
Apr
2020
TLS
SNI
support
enabled
configure
arguments:
--prefix=/etc/nginx
--sbin-path=/usr/sbin/nginx
--conf-path=/etc/nginx/nginx.conf
--pid-path=/run/nginx/nginx.pid
--modules-path=/usr/lib/nginx/modules
--with-file-aio
--with-http_ssl_module
--with-http_v2_module
--with-http_realip_module
--add-module=/tmp/ModSecurity-nginx
--add-module=/tmp/headers-more-nginx-module
--add-module=/tmp/ngx_http_geoip2_module
--add-module=/tmp/nginx_cookie_flag_module
```

As you can see, bunkerized-nginx drops a lot of modules you'd expect from the official images, though I'm not certain it's desirable or intentional.


At a quick glance, one should be able to [compile and drop in the modules](https://www.nginx.com/blog/compiling-dynamic-modules-nginx-plus/) into a preexisting nginx environment, which should allow using the upstream nginx//nginx-unprivileged images as a base.

**Comment by tarrenj** [comment]
I think using the official `alpine:nginx` base image is a good idea! I don't think building it from source the way you've done here adds any extra security, but it does add a bunch of complications and possibility for errors.

Thanks!

**Comment by DistractionRectangle** [comment]
Seconded. As far as missing modules, I personally ran into this over streams, but in general it's probably best to maintain as much compatibility as feasible with the official images from an onboarding perspective.

**Comment by bunkerity** [comment]
Hello @tarrenj  and @DistractionRectangle, fix has been integrated in version [1.1.0](https://github.com/bunkerity/bunkerized-nginx/releases/tag/v1.1.0).

The project now uses nginx:stable-alpine as a base image and additional modules are compiled as dynamic ones. Feel free to open a new issue if needed.
